### ISSUE TITLE
Use buildkite CI

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#12721]
Start using https://buildkite.com/pricing
We should create first PR to use it for "mvn verify".

We can use free first, and then switch to open source pricing

**Comment by Boukhalfa11** [comment]
I am on it

**Comment by romani** [comment]
@ayushactiveat , does buildkite have badges ? For our Readme.md ? To let see status, in case we leak some problems.

**Comment by ayushactiveat** [comment]
@romani yes, buildkite has badges and ig each pipeline gets a unique url which you can see under pipeline->settings → build badges. once checkstyle has its own buildkite org pipeline set up, now for my fork it looks like -(ignore the fail, that is that pre-existing unrelated sevntu) 

<img width="885" height="739" alt="Image" src="https://github.com/user-attachments/assets/2e79a3b5-a6d4-4e8e-88cd-132daa2b9cc7" />
